llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2963,6 +2965,54 @@ class VPScalarIVStepsRecipe : public VPRecipeWithIRFlags {
   }
 };
 
+/// A recipe to represent scalar stores that sink outside the vector loop.
+class VPScalarStoreRecipe : public VPRecipeBase {
+  StoreInst &SI;
```

Comment:
Can this be removed and be replaced by fields that store the needed information independent of StoreInst, so it can also be created without underlying IR?

Reply:
818c8e3bb7420571d4dd2f3fbdfc2ecc1e1eee31
Done, but I can't imagine how it could be used so far. That would be great If you're willing to share your plan.